app/extensions/safe/server-routes/safe.js

```diff
@@ -19,19 +25,60 @@ const safeRoute = {
                 return reply( 'SAFE not connected yet' );
             }
 
-            const data = await app.webFetch( link );
+            if ( headers.range )
+            {
+                const range = headers.range;
+                const rangeArray =
+                  range.substring( BYTES.length, range.length )
+                      .replace( /['"]+/g, '' )
+                      .split( '-' );
+
+                start = rangeArray[0] ? parseInt( rangeArray[0], 10 ) : null;
+                end = rangeArray[1] ? parseInt( rangeArray[1], 10 ) : null;
+
+                //we need to separate out 0- length requests which are not partial content
+                if( start && start !== 0 )
+                {
+                    //should have a start !== 0
+                    isRangeReq = true;
+                }
+
+                if( start === 0 && end )
+                {
+                    isRangeReq = true;
+                }
+            }
+
+            // setup opts object
+            const options = { headers };
+
+            if ( isRangeReq )
+            {
+                options.range = { start, end };
```

Comment:
Since the `Range` header is defined to receive the range end to be inclusive, we'll need to adjust the `end` either here (with `end` + 1 before passing it to `webFetch`) or change the `webFetch` to expect the `end` to be inclusive.

Reply:
I'd opt for this to be baked into webFetch as opposed to client handled. Then we're ready with standard HTTP expected behaviour for other applications.